caput/memh5.py

```diff
@@ -2686,15 +2723,30 @@ def _prepare_compression_args(dset):
             if key != g1.name:
                 # Only create group if we are above the starting level
                 g2.create_group(key)
+
             stack += [entry[k] for k in sorted(entry, reverse=True)]
-        else:  # Is a dataset
+
+        elif key in skip:
+            # Ignore this dataset
+            continue
+
+        elif key in shared:
```

Comment:
I haven't looked at this in depth yet, but I'm worried here that introducing a separate code path for datasets in `shared` means that the last part of the description for the `shared` argument is no longer true: 

>    shared : list, optional
>        List of datasets to share, if `deep_copy_dsets` is True. Otherwise, no effect.

i.e. passing in a `shared` list does seem to have an effect even if `deep_copy_dsets` isn't True.  It seems like, for datasets, in `shared`, there's a lot of stuff in `_prepare_dataset` that isn't happening anymore (like the `to_file` check).  Also `convert_attribute_strings` seems to be ignored for these datasets.

Maybe none of these things are relevant to shared datasets, but if that's the case the documentation needs to better reflect what's actually going on.